### PR TITLE
doc: align the endpoint with environment variable used below to avoid confusing

### PR DESCRIPTION
**Pull request by CarlJi**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
doc: align the endpoint with environment variable used below to avoid confusing

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: docs/csi-dev.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ $ make build
 #### Start CSI driver locally
 ```console
 $ cd $GOPATH/src/github.com/kubernetes-csi/csi-driver-nfs
-$ ./_output/nfsplugin --endpoint tcp://127.0.0.1:10000 --nodeid CSINode -v=5 &
+$ ./_output/nfsplugin --endpoint unix:///tmp/csi.sock --nodeid CSINode -v=5 &
```
Comment (andyzhangx) [inline docs/csi-dev.md:40]: it should be `./bin/nfsplugin ...`

can you also sign the CLA? thanks.

Comment (CarlJi) [reply docs/csi-dev.md]: sure. fixed it now.

**Review by andyzhangx — APPROVED** [review]
/ok-to-test
/lgtm
/approve

**Pull request merged**